pkg/component/kubeproxy/resources.go

```diff
@@ -654,3 +566,128 @@ func (k *kubeProxy) getMode() kubeproxyconfigv1alpha1.ProxyMode {
 	}
 	return "iptables"
 }
+
+func (k *kubeProxy) getInitContainers(kubernetesVersion *semver.Version, image string) []corev1.Container {
+	initContainers := []corev1.Container{
+		{
+			Name:            "cleanup",
+			Image:           image,
+			ImagePullPolicy: corev1.PullIfNotPresent,
+			Command: []string{
+				"sh",
+				"-c",
+				fmt.Sprintf("%s/%s %s", volumeMountPathCleanupScript, dataKeyCleanupScript, volumeMountPathMode),
+			},
+			Env: []corev1.EnvVar{
+				{
+					Name:  "KUBE_PROXY_MODE",
+					Value: string(k.getMode()),
+				},
+				{
+					Name:  "EXECUTE_WORKAROUND_FOR_K8S_ISSUE_109286",
+					Value: strconv.FormatBool(versionutils.ConstraintK8sLess125.Check(kubernetesVersion)),
+				},
+			},
+			SecurityContext: &corev1.SecurityContext{
+				Privileged: pointer.Bool(true),
+			},
+			VolumeMounts: []corev1.VolumeMount{
+				{
+					Name:      volumeNameCleanupScript,
+					MountPath: volumeMountPathCleanupScript,
+				},
+				{
+					Name:      volumeNameKernelModules,
+					MountPath: volumeMountPathKernelModules,
+				},
+				{
+					Name:      volumeNameDir,
+					MountPath: volumeMountPathDir,
+				},
+				{
+					Name:      volumeNameMode,
+					MountPath: volumeMountPathMode,
+				},
+				{
+					Name:      volumeNameKubeconfig,
+					MountPath: volumeMountPathKubeconfig,
+				},
+				{
+					Name:      volumeNameConfig,
+					MountPath: volumeMountPathConfig,
+				},
+			},
+		},
+	}
+
+	k8sGreaterEqual129 := versionutils.ConstraintK8sGreaterEqual129.Check(kubernetesVersion)
+	if k8sGreaterEqual129 {
+		initContainers = append(initContainers, k.getKubeProxyContainer(k8sGreaterEqual129, image, true))
+	}
+
+	return initContainers
+}
+
+func (k *kubeProxy) getKubeProxyContainer(k8sGreaterEqual129 bool, image string, init bool) corev1.Container {
+	container := corev1.Container{
+		Name:            containerName,
+		Image:           image,
+		ImagePullPolicy: corev1.PullIfNotPresent,
+		Command: []string{
+			"/usr/local/bin/kube-proxy",
+			fmt.Sprintf("--config=%s/%s", volumeMountPathConfig, dataKeyConfig),
+			"--v=2",
+		},
+		SecurityContext: &corev1.SecurityContext{
+			Capabilities: &corev1.Capabilities{
+				Add: []corev1.Capability{"NET_ADMIN"},
+			},
```

Comment:
Did you double-checked https://github.com/gardener/gardener/issues/8700#issuecomment-1783358376? Does it work as expected in all cases (ipvs, iptables, etc.) with `NET_ADMIN` only?

Reply:
I actually followed the documentation https://github.com/kubernetes/contributor-site/blob/master/content/en/blog/2024/kube-proxy-non-privileged.md#initializing-kube-proxy-in-an-init-container

Reply:
The pods run without error in both modes with `NET_ADMIN` only. Do we need to run some specific test to find out everything's fine or?

Reply:
The pods were running without error, but there were logs
```
I0111 10:18:24.581074       1 oom_linux.go:82] failed to set "/proc/self/oom_score_adj" to "-999": write /proc/self/oom_score_adj: permission denied
I0111 10:18:24.581143       1 server.go:874] "Failed to apply OOMScore" err="write /proc/self/oom_score_adj: permission denied"
```
Adding `SYS_RESOURCE` solved this. Now both the modes doesn't log any suspicious messages.

Reply:
`SYS_RESOURCE` was also required according to my testing back then. See https://github.com/kubernetes/kubernetes/issues/112171#issuecomment-1253391696. Let's keep it.